Broadcastify Calls - Duplicate Call Handling

Broadcastify handles duplicate calls by generating an error (Broadcastify Metadata Upload Error: SKIPPED---ALREADY-RECEIVED-THIS-CALL). Unfortunately, this error seems to prevent Trunk Recorder from deleting the source audio .wav, the converted m4a, or the metadata.

In a setup where talkgroups are sometimes simulcast, and sometimes not, it's necessary to have the same talkgroups across multiple systems which makes this a fairly common occurrence.

It would be very nice if Trunk Recorder could consider a "SKIPPED" error as a success.

@leee Thanks for taking the time to write the diff. Unfortunately it fails to build - I get the following when I run "rebuild.sh" (manually updated the repo, then ran the diff beforehand):

```
[ 65%] Building CXX object CMakeFiles/recorder.dir/trunk-recorder/uploaders/broadcastify_uploader.cc.o
/home/sdr/trunk-recorder/trunk-recorder/uploaders/broadcastify_uploader.cc: In member function ‘virtual int BroadcastifyUploader::upload(call_data_t*)’:
/home/sdr/trunk-recorder/trunk-recorder/uploaders/broadcastify_uploader.cc:224:14: error: could not convert ‘code.std::__cxx11::basic_string<char>::operator=(((int)((char)((((const char*)"1") != 0) && std::operator==<char, std::char_traits<char>, std::allocator<char> >(message, ((const char*)"SKIPPED---ALREADY-RECEIVED-THIS-CALL"))))))’ from ‘std::__cxx11::basic_string<char>’ to ‘bool’
     if (code = "1" && message == "SKIPPED---ALREADY-RECEIVED-THIS-CALL") {
         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
CMakeFiles/recorder.dir/build.make:158: recipe for target 'CMakeFiles/recorder.dir/trunk-recorder/uploaders/broadcastify_uploader.cc.o' failed
make[2]: *** [CMakeFiles/recorder.dir/trunk-recorder/uploaders/broadcastify_uploader.cc.o] Error 1
CMakeFiles/Makefile2:68: recipe for target 'CMakeFiles/recorder.dir/all' failed
make[1]: *** [CMakeFiles/recorder.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

--- a/trunk-recorder/uploaders/broadcastify_uploader.cc
+++ b/trunk-recorder/uploaders/broadcastify_uploader.cc
@@ -221,6 +221,11 @@ int BroadcastifyUploader::upload(struct call_data_t *call) {
     std::string code = response_buffer.substr(0, spacepos);
     std::string message = response_buffer.substr(spacepos + 1);
 
+    if (code == "1" && message == "SKIPPED---ALREADY-RECEIVED-THIS-CALL") {
+      BOOST_LOG_TRIVIAL(info) << "[" << call->short_name << "]\tTG: " << call->talkgroup << "\tFreq: " << FormatFreq(call->freq) << "\tBroadcastify Upload Skipped: " << message;
+      return 0;
+    }
+
     if (code != "0") {
       BOOST_LOG_TRIVIAL(error) << "[" << call->short_name << "]\tTG: " << call->talkgroup << "\tFreq: " << FormatFreq(call->freq) << "\tBroadcastify Metadata Upload Error: " << message;
       return 1;

